sycl/include/CL/sycl/kernel.hpp

```diff
@@ -8,16 +8,18 @@
 
 #pragma once
 
-#include <CL/sycl/detail/kernel_impl.hpp>
+#include <CL/sycl/detail/common.hpp>
+#include <CL/sycl/info/info_desc.hpp>
 #include <CL/sycl/stl.hpp>
 
-#include <memory>
-
 namespace cl {
 namespace sycl {
 // Forward declaration
 class program;
 class context;
+namespace detail {
+class kernel_impl;
+}
 
 class kernel {
   template <class Obj>
```

Comment:
Suggest moving friend class declarations to the end of the class to align with other classes.